### DISCUSSION TITLE
Scroll to center in GoToDefinition if definition is not in view

### DISCUSSION THREAD

**Discussion by thethongngu** [zed-industries/zed#7651] [Feature Requests]
Current scroll behaviour of GotoDefinition is `Autoscroll::fit`, which is
```
scrolls the minimal amount to (try) and fit all cursors onscreen.
```

Reference: https://github.com/zed-industries/zed/pull/2037

So if:
1. Definition is in the view: no scroll at all (expected behaviour)
2. Definition is not in the view: scroll minimum amount (`Autoscroll::fit`), which means the definition is usually at the bottom of the editor. 

I suggest changing (2) to "scroll to center". I feel it would be more pleasing to our eye and we wouldn't have to scroll up to see the below surrounded code.

This can be applied for search text inside single file too. 

I am thinking about adding another enum into `Autoscroll`, something like `Autoscroll:CenterIfNotVisibleOtherwiseFit` (let me know if you have better name 😄) if it becomes too cumbersome to handle 2 separate `Autoscroll::fit` and `Autoscroll::center`. 

What do you think?

**Comment by louis-she** [comment]
```
Definition is in the view: no scroll at all (expected behaviour)
Definition is not in the view: scroll minimum amount (Autoscroll::fit), which means the definition is usually at the bottom of the editor.
```

This is also vscode does,so for the the vscode migrations like me will be really happy to have this : )

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
